### PR TITLE
Make beatmapset discussion permalinks useful

### LINKED ISSUE #1848: Clicking on permalink does nothing
Clicking permalink doesn't do anything, it has to be right-clicked to copy the link or open in another tab.

screen shot 2017-11-30 at 10 12 48 am

### PR DESCRIPTION
**Pull request by nekodex**
clicking 'permalink' now copies the permalink url to the clipboard

fixes #1848

### CODE REVIEW

File: resources/assets/coffee/react/beatmap-discussions/post.coffee [hunk 1]
```diff
@@ -291,6 +291,18 @@ class BeatmapDiscussions.Post extends React.PureComponent
   permalink: (e) =>
     e.preventDefault()
 
+    hash = BeatmapDiscussionHelper.hash discussionId: @props.discussion.id
+    targetUrl = "#{location.origin}#{location.pathname}#{hash}"
```
Comment (nanaya) [inline resources/assets/coffee/react/beatmap-discussions/post.coffee]: shouldn't it just be `e.target.href`? (which returns full url)

File: resources/assets/coffee/react/beatmap-discussions/post.coffee [hunk 2]
```diff
@@ -291,6 +291,18 @@ class BeatmapDiscussions.Post extends React.PureComponent
...
+
+    # copy url to clipboard
+    clipboard.writeText targetUrl
+
+    # show feedback
+    targetEl = e.currentTarget
+    targetEl.innerHTML = osu.trans('common.buttons.permalink_copied')
+    Timeout.set 2000, ->
+      targetEl.innerHTML = osu.trans('common.buttons.permalink')
```
Comment (nanaya) [inline resources/assets/coffee/react/beatmap-discussions/post.coffee]: save and restore instead? (or react state™ for the correct™ way™)

File: resources/assets/coffee/react/beatmap-discussions/post.coffee [hunk 3]
```diff
@@ -290,6 +290,16 @@ class BeatmapDiscussions.Post extends React.PureComponent
...
+    targetEl = e.currentTarget
+
+    # copy url to clipboard
+    clipboard.writeText targetEl.href
+
+    # show feedback
+    currentLabel = targetEl.innerHTML
+    targetEl.innerHTML = osu.trans('common.buttons.permalink_copied')
+    Timeout.set 2000, ->
+      targetEl.innerHTML = currentLabel
```
Comment (nanaya) [inline resources/assets/coffee/react/beatmap-discussions/post.coffee]: this looks like it'll break when navigating away... except it doesn't ಠ_ಠ

Comment (nekodex) [reply resources/assets/coffee/react/beatmap-discussions/post.coffee]: 🤷‍♀️

**Pull request merged**